docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -1251,6 +1251,17 @@ As the core components use emotion as a styled engine, the props used by emotion
   });
   ```
 
+### `@material-ui/types` `Omit` type
+
+v5 drops the exported `Omit` type in `@material-ui/types`. If you are using that type, replace it with the following:
+
+```ts
+export type DistributiveOmit<T, K extends keyof any> = T extends any ? Omit<T, K> : never;
+```
+
+Notice that the build-in `Omit` type in TypeScript v3.5, while similar, is non-distributive.
+This can cause differences if applied to union types. [See this SO answer for further details](https://stackoverflow.com/a/57103940/1009797).
```

Comment:
How about this changelog for the renamed helper?

```suggestion
- Rename the exported `Omit` type in `@material-ui/types`. The module is now called `DistributiveOmit`. The change was meant to remove the confusion with the built-in `Omit` helper introduced in TypeScript v3.5. This helper while similar is non-distributive. This leads to differences when applied to union types. [See this StackOverflow answer for further details](https://stackoverflow.com/a/57103940/1009797).

  ```diff
  -import { Omit } from '@material-ui/types';
  +import { DistributiveOmit } from '@material-ui/types';
  ```
````